simulator/func_sim/func_sim.h

```diff
@@ -48,6 +57,8 @@ class FuncSim : public Simulator
             PC = target.address;
             sequence_id = target.sequence_id;
         }
+
+        size_t sizeof_register() const final { return sizeof(RegisterUInt); }
```

Comment:
By the way, it needs unit tests as well. With them you would easily spot that fact.